src/builder_state.rs

```diff
@@ -84,17 +80,22 @@ pub struct RequestMessage {
     //pub total_nodes: usize
 }
 /// Response Message to be put on the response channel
-#[derive(Debug, Clone)]
-pub struct ResponseMessage<TYPES: BuilderType> {
-    pub block_hash: BuilderCommitment, //TODO: Need to pull out from hotshot
+#[derive(Debug)]
+pub struct BuildBlockInfo<TYPES: NodeType> {
+    pub builder_hash: BuilderCommitment, //TODO: Need to pull out from hotshot
     pub block_size: u64,
     pub offered_fee: u64,
     pub block_payload: TYPES::BlockPayload,
-    pub metadata: <<TYPES as BuilderType>::BlockPayload as BlockPayload>::Metadata,
-    pub join_handle: Arc<JoinHandle<()>>,
-    pub signature:
-        <<TYPES as BuilderType>::SignatureKey as SignatureKey>::PureAssembledSignatureType,
-    pub sender: TYPES::SignatureKey,
+    pub metadata: <<TYPES as NodeType>::BlockPayload as BlockPayload>::Metadata,
+    pub join_handle: Arc<JoinHandle<<VidScheme as VidSchemeTrait>::Commit>>,
```

Comment:
couldn't this just be `Arc<JoinHandle<vid::VidCommitment>>`?

Reply:
Done